Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -243,27 +276,99 @@ protected override void ProcessRecord()
                 contentStream = new MemoryStream(contentBuffer);
             }
 
+            UploadStreamToGcsObject(contentStream, objContentType, metadataDict, ObjectName);
+        }
+
+        /// <summary>
+        /// Upload a directory to a GCS bucket, aiming to maintain that directory structure as well.
+        /// For example, if we are uploading folder A with file C.txt and subfolder B with file D.txt,
+        /// then the bucket should have A\C.txt and A\B\D.txt
+        /// </summary>
+        private void UploadDirectory(string directory, Dictionary<string, string> metadataDict, string gcsObjectNamePrefix)
+        {
+            if (string.IsNullOrWhiteSpace(directory) || !Directory.Exists(directory))
+            {
+                return;
+            }
+
+            // We have to append / at the end to create a directory on the cloud.
+            string gcsObjectNamePrefixWithSlash = gcsObjectNamePrefix + "/";
+
+            if (TestObjectExists(Service, Bucket, gcsObjectNamePrefixWithSlash) && !Force.IsPresent)
+            {
+                throw new PSArgumentException(
+                    $"Storage object '{gcsObjectNamePrefixWithSlash}' already exists. Use -Force to overwrite.");
+            }
+
+            // Create a directory on the cloud.
+            string objContentType = GetContentType(null, metadataDict, UTF8TextMimeType);
+            byte[] contentBuffer = Encoding.UTF8.GetBytes(Contents);
+            Stream contentStream = new MemoryStream(contentBuffer);
+            UploadStreamToGcsObject(contentStream, objContentType, metadataDict, gcsObjectNamePrefixWithSlash);
+
+            foreach (string file in Directory.EnumerateFiles(directory))
+            {
+                // Upload each individual file.
+                string fileName = Path.GetFileName(file);
+                string fileWithGcsObjectNamePrefix = Path.Combine(gcsObjectNamePrefix, fileName);
+                // We have to replace \ with / so it will be created with correct folder structure.
+                fileWithGcsObjectNamePrefix = ConvertLocalToGcsFolderPath(fileWithGcsObjectNamePrefix);
+                UploadStreamToGcsObject(new FileStream(file, FileMode.Open),
```

Comment:
Line breaking rules are hard. But I believe you should put `new FileStream` on a new line, rather than putting only one of the parameters to `UploadStreamtoGcsObject` on the same line.